subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/JavaInstallationSpec.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Allows managing Java installations to be used in a build.
+ * <p>A Java installation may contain a JDK, or a JRE, or both.</p>
+ * <p>An instance of this service is available for injection into tasks, plugins and other types.
+ *
+ * @since 6.6
+ */
+@Incubating
+public interface JavaInstallationSpec {
+
+    /**
+     * Registers an existing local java installation.
+     *
+     * <p>The directory is evaluated as per {@code Project.file(Object)}</p>
+     */
+    void registerExisting(String name, Object path);
```

Comment:
:o: Why using `Object` here? It seems we should be using `DirectoryProvider` now.